openssh/README.md

```diff
@@ -0,0 +1,106 @@
+
+For example, with this build command
+```bash
+docker build --build-arg LIBOQS_BUILD_DEFINES="-DOQS_USE_CPU_EXTENSIONS=OFF" -f Dockerfile -t oqs-openssh-generic .
+``` 
+a generic system without processor-specific runtime optimizations is built, thus ensuring execution on all computers (at the cost of maximum runtime performance).
+
+## Usage
+
+Information how to use the image is [available in the separate file USAGE.md](USAGE.md).
+
+## Build options
+
+The Dockerfile provided allows for some customization of the image built. Those build arguments can be used at buildtime via the flag `--build-arg`, e.g. `docker build --build-arg INSTALL_DIR="/some/directory/" -t myimage .`.
+
+### INSTALL_DIR
+
+This defines the resultant location of the software installation.
+
+By default this is `/opt/oqs-ssh`. When it is changed, every occurrence of this default path is replaced with it at build time. That means that for example the `ssh_config` file copied to the container can differ from the original [ssh_config](ssh_config) because it is edited during build.
+
+### LIBOQS_BUILD_DEFINES
+
+This permits changing the build options for the underlying library with the quantum safe algorithms. All possible options are documented [here](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs).
+
+By default, the image is built such as to have maximum portability regardless of CPU type and optimizations available, i.e. to run on the widest possible range of cloud machines.
+
+### OPENSSH_BUILD_OPTIONS
+
+This allows to configure some additional build options for building OQS-OpenSSH. Those options, if specified, will be appended to the `./configure` command as shown [here](https://github.com/open-quantum-safe/openssh#step-2-build-the-fork). Some parameters are configured as follows in the Dockerfile as they are essential to the build:
+```sh
+./configure \
+    --with-libs=-lm \
+    --prefix=${INSTALL_DIR} \
+    --sysconfdir=${INSTALL_DIR} \
+    --with-liboqs-dir=/opt/ossh-src/oqs \
+    --with-mantype=man \
+    ${OPENSSH_BUILD_OPTIONS}
+```
+These parameters will be overridden if specified again in `OPENSSH_BUILD_OPTIONS`.
+
+`/opt/ossh-src/oqs` is the location the intermediate build in the Dockerfile writes the compiled liboqs binaries to. This should not be changed, as it does not influence the final docker image.
+### MAKE_DEFINES
+
+Allow setting parameters to `make` operation, e.g., `-j nnn` where nnn defines the number of jobs run in parallel during build. 
+
+The default is conservative and known not to overload normal machines (default: `-j 2`). If one has a very powerful (many cores, >64GB RAM) machine, passing larger numbers (or only `-j` for maximum parallelism) speeds up the build considerably.
+
+### OQS_USER
```

Comment:
Any plans of supporting key-based client auth?

Reply:
It actually is supported, like with standard ssh you may add your public identity key (that is generated in `key-regen.sh` based on `ssh_config`)  to the server's `known_keys` file.

Might be worth mentioning in the docs though.